### PR TITLE
[ISSUE-622] Improve logging in scheduler-extender

### LINKED ISSUE #622: Improve logging in csi-baremetal-extender
**Is your feature request related to a problem? Please describe.**
When ACR creating while filtering, we can't verify a reason correctly. 

**Describe the solution you'd like**
Need to add log massages for each check

### PR DESCRIPTION
**Pull request by safronovD**
### Resolves #622 

- Add log on capacity request creating
- Add info about PVC
- Add info about PVC skipping

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
```
Nov 24 16:06:03.239 [INFO] [Extender] [FilterHandler] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Processing request: &{POST /filter HTTP/1.1 1 1 map[Accept-Encoding:[gzip] Content-Length:[12810] Content-Type:[application/json] User-Agent:[Go-http-client/1.1]] 0xc0004b9900 <nil> 12810 [] false 127.0.0.1:8889 map[] map[] <nil> map[] 127.0.0.1:46988 /filter <nil> <nil> <nil> 0xc0004b9940}
Nov 24 16:06:03.245 [INFO] [Extender] [FilterHandler] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Filtering
Nov 24 16:06:03.247 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] SC map: map[csi-baremetal-sc:ANY csi-baremetal-sc-hdd:HDD csi-baremetal-sc-hddlvg:HDDLVG csi-baremetal-sc-nvme:NVME csi-baremetal-sc-nvme-raw-part:NVME csi-baremetal-sc-nvmelvg:NVMELVG csi-baremetal-sc-ssd:SSD csi-baremetal-sc-ssdlvg:SSDLVG csi-baremetal-sc-syslvg:SYSLVG]
Nov 24 16:06:03.249 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Volume logs details: {Name:logs VolumeSource:{HostPath:nil EmptyDir:nil GCEPersistentDisk:nil AWSElasticBlockStore:nil GitRepo:nil Secret:nil NFS:nil ISCSI:nil Glusterfs:nil PersistentVolumeClaim:&PersistentVolumeClaimVolumeSource{ClaimName:logs-web-1,ReadOnly:false,} RBD:nil FlexVolume:nil Cinder:nil CephFS:nil Flocker:nil DownwardAPI:nil FC:nil AzureFile:nil ConfigMap:nil VsphereVolume:nil Quobyte:nil AzureDisk:nil PhotonPersistentDisk:nil Projected:nil PortworxVolume:nil ScaleIO:nil StorageOS:nil CSI:nil}}
Nov 24 16:06:03.265 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] PVC logs-web-1 status: {Phase:Pending AccessModes:[] Capacity:map[] Conditions:[]} spec: {AccessModes:[ReadWriteOnce] Selector:nil Resources:{Limits:map[] Requests:map[storage:{i:{value:62914560 scale:0} d:{Dec:<nil>} s:60Mi Format:BinarySI}]} VolumeName: StorageClassName:0xc000434670 VolumeMode:0xc000434680 DataSource:nil} SC: csi-baremetal-sc
Nov 24 16:06:03.270 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Volume www details: {Name:www VolumeSource:{HostPath:nil EmptyDir:nil GCEPersistentDisk:nil AWSElasticBlockStore:nil GitRepo:nil Secret:nil NFS:nil ISCSI:nil Glusterfs:nil PersistentVolumeClaim:&PersistentVolumeClaimVolumeSource{ClaimName:www-web-1,ReadOnly:false,} RBD:nil FlexVolume:nil Cinder:nil CephFS:nil Flocker:nil DownwardAPI:nil FC:nil AzureFile:nil ConfigMap:nil VsphereVolume:nil Quobyte:nil AzureDisk:nil PhotonPersistentDisk:nil Projected:nil PortworxVolume:nil ScaleIO:nil StorageOS:nil CSI:nil}}
Nov 24 16:06:03.271 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] PVC www-web-1 status: {Phase:Pending AccessModes:[] Capacity:map[] Conditions:[]} spec: {AccessModes:[ReadWriteOnce] Selector:nil Resources:{Limits:map[] Requests:map[storage:{i:{value:62914560 scale:0} d:{Dec:<nil>} s:60Mi Format:BinarySI}]} VolumeName: StorageClassName:0xc000434750 VolumeMode:0xc000434760 DataSource:nil} SC: csi-baremetal-sc
Nov 24 16:06:03.271 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Volume data details: {Name:data VolumeSource:{HostPath:nil EmptyDir:nil GCEPersistentDisk:nil AWSElasticBlockStore:nil GitRepo:nil Secret:nil NFS:nil ISCSI:nil Glusterfs:nil PersistentVolumeClaim:&PersistentVolumeClaimVolumeSource{ClaimName:data-web-1,ReadOnly:false,} RBD:nil FlexVolume:nil Cinder:nil CephFS:nil Flocker:nil DownwardAPI:nil FC:nil AzureFile:nil ConfigMap:nil VsphereVolume:nil Quobyte:nil AzureDisk:nil PhotonPersistentDisk:nil Projected:nil PortworxVolume:nil ScaleIO:nil StorageOS:nil CSI:nil}}
Nov 24 16:06:03.271 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] PVC data-web-1 status: {Phase:Pending AccessModes:[] Capacity:map[] Conditions:[]} spec: {AccessModes:[ReadWriteOnce] Selector:nil Resources:{Limits:map[] Requests:map[storage:{i:{value:62914560 scale:0} d:{Dec:<nil>} s:60Mi Format:BinarySI}]} VolumeName: StorageClassName:0xc000434870 VolumeMode:0xc000434890 DataSource:nil} SC: csi-baremetal-sc
Nov 24 16:06:03.271 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Volume default-token-x5fv5 details: {Name:default-token-x5fv5 VolumeSource:{HostPath:nil EmptyDir:nil GCEPersistentDisk:nil AWSElasticBlockStore:nil GitRepo:nil Secret:&SecretVolumeSource{SecretName:default-token-x5fv5,Items:[]KeyToPath{},DefaultMode:*420,Optional:nil,} NFS:nil ISCSI:nil Glusterfs:nil PersistentVolumeClaim:nil RBD:nil FlexVolume:nil Cinder:nil CephFS:nil Flocker:nil DownwardAPI:nil FC:nil AzureFile:nil ConfigMap:nil VsphereVolume:nil Quobyte:nil AzureDisk:nil PhotonPersistentDisk:nil Projected:nil PortworxVolume:nil ScaleIO:nil StorageOS:nil CSI:nil}}
Nov 24 16:06:03.272 [DEBU] [Extender] [FilterHandler] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Required capacity: [Name:"logs-web-1" StorageClass:"ANY" Size:62914560  Name:"www-web-1" StorageClass:"ANY" Size:62914560  Name:"data-web-1" StorageClass:"ANY" Size:62914560 ]
Nov 24 16:06:03.316 [INFO] [Extender] [FilterHandler] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] Construct response. Get 1 nodes in request. Among them suitable nodes count is 1. Filtered out nodes - map[]
```

### CODE REVIEW

File: pkg/scheduler/extender/extender.go [hunk 1]
```diff
@@ -253,15 +255,20 @@ func (e *Extender) gatherCapacityRequestsByProvisioner(ctx context.Context, pod
 				return nil, baseerr.ErrorNotFound
 			}
 
-			ll.Debugf("PVC details: %s", pvc.String())
+			ll.Debugf("PVC %s status: %+v", pvc.Name, pvc.Status)
+			ll.Debugf("PVC %s spec: %+v", pvc.Name, pvc.Spec)
+			ll.Debugf("PVC %s SC: %s", pvc.Name, *pvc.Spec.StorageClassName)
 
 			if pvc.Spec.StorageClassName == nil {
+				ll.Infof("PVC %s skipped due to empty StorageClass", pvc.Name)
```
Comment (mishoyama) [inline pkg/scheduler/extender/extender.go:263]: Warning?

File: pkg/scheduler/extender/extender.go [hunk 2]
```diff
@@ -253,15 +255,20 @@ func (e *Extender) gatherCapacityRequestsByProvisioner(ctx context.Context, pod
...
 				continue
 			}
-			if _, ok := scs[*pvc.Spec.StorageClassName]; !ok {
+			if pvc.Status.Phase == coreV1.ClaimBound {
+				ll.Infof("PVC %s is Bound", pvc.Name)
 				continue
 			}
-			if pvc.Status.Phase == coreV1.ClaimBound || pvc.Status.Phase == coreV1.ClaimLost {
+			if pvc.Status.Phase == coreV1.ClaimLost {
+				ll.Infof("PVC %s is Lost", pvc.Name)
```
Comment (mishoyama) [inline pkg/scheduler/extender/extender.go:271]: Warning?

File: pkg/scheduler/extender/extender.go [hunk 3]
```diff
@@ -253,15 +255,20 @@ func (e *Extender) gatherCapacityRequestsByProvisioner(ctx context.Context, pod
 				return nil, baseerr.ErrorNotFound
 			}
 
-			ll.Debugf("PVC details: %s", pvc.String())
+			ll.Debugf("PVC %s status: %+v", pvc.Name, pvc.Status)
+			ll.Debugf("PVC %s spec: %+v", pvc.Name, pvc.Spec)
```
Comment (mishoyama) [inline pkg/scheduler/extender/extender.go:259]: Won't it print SC name from spec as well?

File: pkg/scheduler/extender/extender.go [hunk 4]
```diff
@@ -253,15 +255,20 @@ func (e *Extender) gatherCapacityRequestsByProvisioner(ctx context.Context, pod
 				return nil, baseerr.ErrorNotFound
 			}
 
-			ll.Debugf("PVC details: %s", pvc.String())
+			ll.Debugf("PVC %s status: %+v", pvc.Name, pvc.Status)
```
Comment (mishoyama) [inline pkg/scheduler/extender/extender.go:258]: Can we have just one log statement here if all of them have similar severity level?

Comment (mishoyama) [reply pkg/scheduler/extender/extender.go:258]: BTW shouldn't it be INFO?

File: pkg/scheduler/extender/extender.go (hunk 1 above)
Comment (safronovD) [reply pkg/scheduler/extender/extender.go]: fixed

File: pkg/scheduler/extender/extender.go (hunk 2 above)
Comment (safronovD) [reply pkg/scheduler/extender/extender.go]: fixed

File: pkg/scheduler/extender/extender.go (hunk 3 above)
Comment (safronovD) [reply pkg/scheduler/extender/extender.go]: It is a pointer - `StorageClassName:0xc000434670`

File: pkg/scheduler/extender/extender.go (hunk 4 above)
Comment (safronovD) [reply pkg/scheduler/extender/extender.go]: fixed

Comment (safronovD) [reply pkg/scheduler/extender/extender.go]: I think it's too much for info
```
Nov 24 16:06:03.271 [DEBU] [Extender] [gatherCapacityRequestsByProvisioner] [web-1] [c75ace5c-24c6-42cc-81cc-7e60412aeee6] PVC www-web-1 status: {Phase:Pending AccessModes:[] Capacity:map[] Conditions:[]} spec: {AccessModes:[ReadWriteOnce] Selector:nil Resources:{Limits:map[] Requests:map[storage:{i:{value:62914560 scale:0} d:{Dec:<nil>} s:60Mi Format:BinarySI}]} VolumeName: StorageClassName:0xc000434750 VolumeMode:0xc000434760 DataSource:nil} SC: csi-baremetal-sc
```
@rkiselenko? @yurkov-anton?

**Review by mishoyama — APPROVED** [review]

File: pkg/scheduler/extender/extender.go (hunk 4 above)
Comment (mishoyama) [reply pkg/scheduler/extender/extender.go]: Do we really need to print all PVC details or only those which we are checking?

Comment (safronovD) [reply pkg/scheduler/extender/extender.go]: I reduce log massage to print only spec and status. We can have invalid PVC (for example: accessMode or Limits) and it's better to store it here, if PVC is deleted or changed

**Review by rkiselenko — APPROVED** [review]

**Pull request merged**